server/service/devices.go

```diff
@@ -454,6 +455,81 @@ func (svc *Service) GetTransparencyURL(ctx context.Context) (string, error) {
 	return transparencyURL, nil
 }
 
+// ///////////////////////////////////////////////////////////////////////////////
+// Software title icons
+// ///////////////////////////////////////////////////////////////////////////////
+type getDeviceSoftwareIconRequest struct {
+	Token           string `url:"token"`
+	SoftwareTitleID uint   `url:"software_title_id"`
+}
+
+func (r *getDeviceSoftwareIconRequest) deviceAuthToken() string {
+	return r.Token
+}
+
+type getDeviceSoftwareIconResponse struct {
+	Err         error  `json:"error,omitempty"`
+	ImageData   []byte `json:"-"`
+	ContentType string `json:"-"`
+	Filename    string `json:"-"`
+	Size        int64  `json:"-"`
+}
+
+func (r getDeviceSoftwareIconResponse) Error() error { return r.Err }
+
+type getDeviceSoftwareIconRedirectResponse struct {
+	RedirectURL string `json:"-"`
+}
+
+func (r getDeviceSoftwareIconRedirectResponse) Error() error { return nil }
+
+func (r getDeviceSoftwareIconRedirectResponse) HijackRender(ctx context.Context, w http.ResponseWriter) {
+	w.Header().Set("Location", r.RedirectURL)
+	w.WriteHeader(http.StatusTemporaryRedirect)
```

Comment:
```suggestion
	w.WriteHeader(http.StatusFound)
```

Also, why does this look different than `getSoftwareTitleIconsRedirectResponse`?